### PR TITLE
Known null safety issues

### PR DESCRIPTION
**Pull request by mit-mit**
Add a new section with known null safety issues.

Corresponding Flutter PR: https://github.com/flutter/website/pull/5284

### CODE REVIEW

**Comment by mit-mit** [comment]
cc @kevmoo can you add what you know off? For example, I think some parts of build-and-test don't work yet?

**Comment by kevmoo** [comment]
> cc @kevmoo can you add what you know off? For example, I think some parts of build-and-test don't work yet?

we've been focused on things needed by app runtime. Build and test code needed for app runtime has been migrated.

For instance, you can use json_serializable to build null-safe code. Works great. But you cannot run json_serialiable builder null safe (yet).

Clear as mud?

**Comment by mit-mit** [comment]
I'm not seeing the mud entirely clear :-). Mayve @kevmoo you can PR the PR?

**Comment by kevmoo** [comment]
> I'm not seeing the mud entirely clear :-). Mayve @kevmoo you can PR the PR?

I don't have anything off of the top of my head to add. @natebosch @jakemac53 ?

File: src/null-safety/index.md [hunk 1]
```diff
@@ -334,6 +334,18 @@ If you find any issues with null safety please [give us feedback.][]
...
 
+## Known issues
+
+Not all parts of the Dart SDK support null safety yet,
+as some parts still need additional work to
+[migrate to null safety](https://dart.dev/null-safety/migration-guide).
+
+We're currently aware of the following issues:
+
+  * Migration of the pub.dev packages owned by the Dart team
```
Comment (kwalrath) [inline src/null-safety/index.md:345]: I wouldn't call these package part of the SDK. Should we broaden the intro?

File: src/null-safety/index.md [hunk 2]
```diff
@@ -334,6 +334,18 @@ If you find any issues with null safety please [give us feedback.][]
 
 [language version]: /guides/language/evolution#language-versioning
 
+## Known issues
+
+Not all parts of the Dart SDK support null safety yet,
+as some parts still need additional work to
+[migrate to null safety](https://dart.dev/null-safety/migration-guide).
```
Comment (kwalrath) [inline src/null-safety/index.md:341]:
```suggestion
[migrate to null safety](/null-safety/migration-guide).
```

File: src/null-safety/index.md [hunk 3]
```diff
@@ -334,6 +334,18 @@ If you find any issues with null safety please [give us feedback.][]
...
+
+We're currently aware of the following issues:
+
+  * Migration of the pub.dev packages owned by the Dart team
+    is nearly complete, but a few are still missing. See pub.dev for
+    [the current list](https://pub.dev/packages?q=publisher%3Adart.dev&null-safe=1).
```
Comment (kwalrath) [inline src/null-safety/index.md:347]: It isn't clear what "the current list" is referring to.

```suggestion
    [null-safe packages from the Dart team][ns-dart-pkgs].

[ns-dart-pkgs]: {{site.pub-pkg}}?q=publisher%3Adart.dev&null-safe=1
```

File: src/null-safety/index.md (hunk 1 above)
Comment (mit-mit) [reply src/null-safety/index.md:345]: Sure. Dart platform? Or just Dart?

File: src/null-safety/index.md [hunk 4]
```diff
@@ -334,6 +334,18 @@ If you find any issues with null safety please [give us feedback.][]
 
 [language version]: /guides/language/evolution#language-versioning
 
+## Known issues
+
+Not all parts of the Dart SDK support null safety yet,
+as some parts still need additional work to
```
Comment (kwalrath) [inline src/null-safety/index.md:340]: Maybe this (and then we can modify "issues" below to make clear that we're only going to talk about the Dart team's stuff):

```suggestion
Some parts of the Dart ecosystem still need additional work to
```

File: src/null-safety/index.md [hunk 5]
```diff
@@ -334,6 +334,18 @@ If you find any issues with null safety please [give us feedback.][]
...
+[migrate to null safety](https://dart.dev/null-safety/migration-guide).
+
+We're currently aware of the following issues:
```
Comment (kwalrath) [inline src/null-safety/index.md:343]:
```suggestion
The Dart team is currently aware of the following issues:
```

File: src/null-safety/index.md (hunk 1 above)
Comment (kwalrath) [reply src/null-safety/index.md:345]: I made a couple of suggestions above that might help

**Comment by mit-mit** [comment]
@kwalrath, I'll land this now, please do followup PRs if additional tweaking is needed

**Pull request merged**